Fix focus state on licence-finder journey

maxgds opened the pull request:
https://trello.com/c/uJOqVUmd/83-update-license-finder-to-govukpublishingcomponents-v21-with-compatibility-mode-on

This PR tackles improving the focus state throughout the licence finder journey starting at 
https://www.gov.uk/licence-finder/sectors

This involves:
Updating the components gem and removing legacy mixins
Adding the govuk-link class to various links
Replacing the buttons with component versions (and adjusting the js to accommodate this)
Tweaking the layout where needed to support the focus state

*Out of scope for this work*
Improving the strange design choices - odd layout including spurious padding has been mostly left as it was
Updating the CSS to avoid element selectors
Replacing form elements with components

### Before/after screenshots
![Screenshot 2019-11-27 at 10 22 30](https://user-images.githubusercontent.com/31649453/69715253-dd13f900-10ff-11ea-9b48-830e4774b254.png)
![Screenshot 2019-11-27 at 10 22 12](https://user-images.githubusercontent.com/31649453/69715265-e309da00-10ff-11ea-9e73-cb386979ff1e.png)

-----

![Screenshot 2019-11-27 at 10 23 30](https://user-images.githubusercontent.com/31649453/69715356-07fe4d00-1100-11ea-975e-e7ed6f74aa0c.png)
![Screenshot 2019-11-27 at 10 23 39](https://user-images.githubusercontent.com/31649453/69715357-0896e380-1100-11ea-85dd-65bfceb6312c.png)
-----
![Screenshot 2019-11-27 at 10 24 30](https://user-images.githubusercontent.com/31649453/69715460-3419ce00-1100-11ea-9259-a37652f15069.png)
![Screenshot 2019-11-27 at 10 24 38](https://user-images.githubusercontent.com/31649453/69715462-34b26480-1100-11ea-9c6d-7dc0ab76fe56.png)
-----
![Screenshot 2019-11-27 at 11 06 43](https://user-images.githubusercontent.com/31649453/69718374-18192b00-1106-11ea-8f30-8f3f24d0596c.png)
![Screenshot 2019-11-27 at 11 06 53](https://user-images.githubusercontent.com/31649453/69718375-18b1c180-1106-11ea-9c0c-248e6f08f481.png)
-----
![Screenshot 2019-11-27 at 11 08 08](https://user-images.githubusercontent.com/31649453/69718428-40088e80-1106-11ea-8463-721633a25d61.png)
![Screenshot 2019-11-27 at 11 08 16](https://user-images.githubusercontent.com/31649453/69718429-40088e80-1106-11ea-9c71-413d3f4dbe7a.png)
-----
![Screenshot 2019-11-27 at 11 10 12](https://user-images.githubusercontent.com/31649453/69718544-8d84fb80-1106-11ea-83ea-bc8d5a2983e2.png)
![Screenshot 2019-11-27 at 11 10 22](https://user-images.githubusercontent.com/31649453/69718545-8d84fb80-1106-11ea-8b4b-0326ddb2e613.png)
-----
![Screenshot 2019-11-27 at 11 11 06](https://user-images.githubusercontent.com/31649453/69718587-a7bed980-1106-11ea-9a98-43498976d818.png)
![Screenshot 2019-11-27 at 11 11 15](https://user-images.githubusercontent.com/31649453/69718588-a8577000-1106-11ea-8e71-c0a64c2e3346.png)
-----

maxgds commented on app/views/licence_finder/activities.html.erb:
```diff
@@ -30,11 +30,14 @@
-      </div>
-    <% end %>
+
+    <div class="button-container js-hidden"><!-- FIXME - what about non-js? -->
```
oops, need to kill that comment

maxgds commented on app/views/licence_finder/sectors.html.erb:
```diff
@@ -6,7 +6,9 @@
       <%= hidden_field_tag "sectors", params[:sectors], id: "hidden-sectors" %>
       <label class="visuallyhidden" for="search-sectors">Search for areas</label>
       <%= text_field_tag "q", params[:q], id: "search-sectors" %>
```
Yeah, I thought the remit was to make links work and address other components later, though at this stage? Do you think I need to do both here?

injms replied:
I think that it's worth making the new focus states work on all elements that need it, including input elements. Whether that's by using the input component from the gem or by adding a `govuk-` class to the input doesn't matter. 🙂

maxgds commented on spec/support/custom_matchers.rb:
```diff
@@ -57,7 +57,9 @@ def check_add_remove_link(type, label)
-    expect(element).to have_xpath("a[@class = '#{type_class}']")
+
+    expect(element).to have_selector(:css, "a.govuk-link.#{type_class}")
+    #expect(element).to have_xpath("a[@class = 'govuk-link #{type_class}']")
```
ugh, and this one needs removing

maxgds commented:
@injms I've addressed all issues, including replacing the form input and the form dropdown with components, please take another look when you have time.
